src/use-swr.ts

```diff
@@ -394,7 +394,7 @@ function useSWR<Data = any, Error = any>(
     const softRevalidate = () => revalidate({ dedupe: true })
 
     // trigger a revalidation
-    if (!config.initialData) {
+    if (!config.initialData || !config.autoLoad) {
```

Comment:
```suggestion
    if (!config.initialData && config.autoLoad) {
```

According to your use case, we should require `autoLoad` to be `true` to start revalidation at the beginning.

Reply:
yep sorry, I inverted condition